python/pyarrow/includes/libarrow.pxd

```diff
@@ -3026,3 +3023,34 @@ cdef extern from "arrow/python/udf.h" namespace "arrow::py" nogil:
 
 cdef extern from "arrow/compute/cast.h" namespace "arrow::compute":
     CResult[CDatum] Cast(const CDatum& value, const CCastOptions& options)
+
+cdef extern from "arrow/type_traits.h" namespace "arrow":
+    c_bool c_is_integer "is_integer"(Type type_id)
```

Comment:
@AlenkaF Thank you for your review and advice.

Inspired by your suggestions, I further explored the code and made some modifications.

Could this be what @pitrou intended?

Reply:
Looks great to me!
The failures do not seem connected, not sure there are open issues for them though. Will look into it.

Reply:
Opened https://github.com/apache/arrow/issues/41563 for the failing builds